ci/docker-compose.yml

```diff
@@ -117,6 +117,16 @@ services:
       - ..:/risingwave
 
   release-env:
+    # build binaries on a earlier Linux distribution (therefore with earlier version GLIBC)
+    # See https://github.com/risingwavelabs/risingwave/issues/4556 for more details.
+    #
+    # GLIBC versions on some systems:
+    # Amazon Linux 2: 2.26 (EOL 2025-06-30) (We will definitely want to support this)
+    # AL2023: 2.34
+    # Ubuntu 18.04: 2.27 (Already EOL 2023-05-31)
+    # Ubuntu 20.04: 2.31
+    #
+    # manylinux2014: CentOS 7 (EOL 2024-06-30), GLIBC 2.17
```

Comment:
Just adding a link for future (in 7 months) reference:
https://github.com/pypa/manylinux
> The goal of the manylinux project is to provide a convenient way to distribute binary Python extensions as wheels on Linux. This effort has produced [PEP 513](https://www.python.org/dev/peps/pep-0513/) (manylinux1), [PEP 571](https://www.python.org/dev/peps/pep-0571/) (manylinux2010), [PEP 599](https://www.python.org/dev/peps/pep-0599/) (manylinux2014) and [PEP 600](https://www.python.org/dev/peps/pep-0600/) (manylinux_x_y).
>
> PEP 599 defines the following platform tags:
> * `manylinux2014_x86_64 `
> * ...
>
> Wheels are built on CentOS 7 which will reach End of Life (EOL) on June 30th, 2024.
>
> PEP 600 has been designed to be "future-proof" and does not enforce specific symbols and a specific distro to build. It only states that a wheel tagged `manylinux_x_y` shall work on any distro based on `glibc>=x.y`. The manylinux project supports:
> * `manylinux_2_24` images for `x86_64`, `i686`, `aarch64`, `ppc64le` and `s390x`.
> * `manylinux_2_28` images for `x86_64`, `aarch64`, `ppc64le` and `s390x`.

Reply:
very unfortunately 2_24 is already EOL. When 2014 (2_17) is EOL, we won’t have a nice alternative immediately. 🤡

2_28 is larger than Amazon Linux. And... Amazon Linux‘s EOL is sooooo long